### DISCUSSION TITLE
Is there a way to change the sequence of certain columns in an ReplicatedMergeTree

### DISCUSSION THREAD

**Discussion by mo-avatar** [ClickHouse/ClickHouse#37097] [Q&A] [answered]
Is there a way to change the sequence of certain columns in an ReplicatedMergeTree?

**Comment by UnamedRus** [comment, marked as answer]
Did you check

```ALTER TABLE table MODIFY COLUMN xxx AFTER yyy;```

**Comment by flyly0755** [comment]
with MergeTree table, i take a try:
ALTER TABLE rootCauseExtra MODIFY COLUMN columnx AFTER COLUMN columny;
ALTER TABLE rootCauseExtra MODIFY COLUMN columnx AFTER columny;
all failed, with error info
ClickHouse exception, code: 1002, host: xxx, port: 8123; Code: 62. DB::Exception: Syntax error: failed at position 56 ('COLUMN'): COLUMN remark. Expected one of: DEFAULT, MATERIALIZED, ALIAS, COMMENT, CODEC, TTL, REMOVE, FIRST, AFTER, token, Comma, INTO OUTFILE, FORMAT, SETTINGS, end of query. (SYNTAX_ERROR) (version 21.12.3.32 (official build))
I think this is general function not related with table engine type.
any issue with my sql language?

**Comment by den-crane** [comment]
@flyly0755 see https://clickhouse.com/docs/en/sql-reference/statements/alter/column/#modify-column

```sql
CREATE TABLE users (
    c1 Int16,
    c2 String
) ENGINE = MergeTree
ORDER BY c1;

DESCRIBE users;
┌─name─┬─type───┬
│ c1   │ Int16  │
│ c2   │ String │
└──────┴────────┴

ALTER TABLE users MODIFY COLUMN c2 String FIRST;

DESCRIBE users;
┌─name─┬─type───┬
│ c2   │ String │
│ c1   │ Int16  │
└──────┴────────┴

ALTER TABLE users ALTER COLUMN c2 TYPE String AFTER c1;

DESCRIBE users;
┌─name─┬─type───┬
│ c1   │ Int16  │
│ c2   │ String │
└──────┴────────┴
```

**Reply by flyly0755** [reply]
thanks, with datatype it's ok, but I think datatype can be a optional parameter, which is more robust.
if only change column sequence, so with 
**ALTER TABLE rootCauseExtra MODIFY COLUMN columnx fisrt**  and so on.
and if at the same time change the colomn datatype, then with 
**ALTER TABLE rootCauseExtra MODIFY COLUMN columnx string first** and so on.

**ALTER TABLE rootCauseExtra MODIFY COLUMN columnx [string] first**  the parameter in [] as a optional.
Which can be better:)

**Comment by den-crane** [comment]
>I think datatype can be a optional parameter, which is more robust.

Everybody agree with you.
But unfortunately, it's not optional.
